# BIMM-181
Here's the code from my Bioinformatics Algorithms class. In this repository you'll find plenty of algorithms for pattern matching, assembly, alignment, chromosome breakpoint computation, clustering, and more.

A non-exhaustive list of topics follows:
 - Finding origins of replication 
  - exact & approximate pattern matching for short genomes
  - computation of genomic skew
 - Motif finding
  - Probabilistic & information-theoretic criteria for identifying motifs
  - Greedy motif finding
  - Random motif finding/Gibbs sampler methods
 - Genome assembly
  - Overlap graphs & Hamiltonian paths
  - De Bruijn graphs & Eulerian paths
  - Repeat graphs, long reads, and depth/coverage
  - Paired reads and De Bruijn graphs
 - Sequence alignment
  - Global v. local alignment & other alignment schemes
  - Alignment tables, topological orderings, and table initialization for different alignment schemes
  - Linear and affine penalties for alignment (changing graph connectivity)
  - Backtracking
  - Linear memory alignment
  - Multiple sequence alignments
 - Chromosome breakage & rearrangements
  - Greedy permutation sort
  - Permutation reversal sort
  - Breakpoints in chromosomes and breakpoint elimination
  - Breakpoint graphs & 2-Break distance
 - Clustering
  - k-Means clustering & the Lloyd Algorithm
   - initialization methods
  - soft k-Means clustering
  - hierarchal clustering
 - Finding sequence matches in the genome
  - Exact & approximate pattern matching for long genomes
  - Suffix Arrays
  - Tries, Trees, and the Burrows Wheeler transform
